compiler/infra/OMRCfg.cpp

```diff
@@ -166,48 +166,68 @@ OMR::CFG::addExceptionEdge(
       TR::CFGNode *f,
       TR::CFGNode *t)
    {
+   addExceptionEdgeInternal(f, t, true);
+   }
+
+void
+OMR::CFG::addExceptionEdgeUnchecked(
+      TR::CFGNode *f,
+      TR::CFGNode *t)
+   {
+   addExceptionEdgeInternal(f, t, false);
+   }
+
+void
+OMR::CFG::addExceptionEdgeInternal(
```

Comment:
couldn't 233-246 (the actual add) be the implementation of addExceptionEdgeUnchecked with the checking logic in addExceptionEdge and a call to addExceptionEdgeUnchecked at the end to do the actual add? I'm not sure what is gained by keeping the implementation in its own function with a hard to read Boolean flag. Is there a complication you encountered that I might be missing that drove this particular design?

Reply:
The reason I did it this way was so that the trace message and the assert (which happen before the check for duplicate exception edges) would be done in both the checked and unchecked versions. I guess it's probably not a big deal to duplicate the trace message. Should the assert also be done in the unchecked version or just in the checked version?

I agree it would be cleaner to not have this boolean flag.

Reply:
So the assert for hasSuccessor should always run. The trace about adding an edge should always run so I think you can do the deduplicate easily right?